### PR TITLE
smart forms

### PR DESCRIPTION
**Pull request by mo-fc**

### CODE REVIEW

File: source/docs/appendices/migration-guides.md [hunk 1]
```diff
@@ -87,6 +87,58 @@ query PayzenEmbeddedQuery {
 
 and ensure that the `modules/Checkout/Payment/AdditionalPaymentInformation/PayzenEmbeddedForm/PayzenScriptWrapper.js` is not overridden either (very unlikely).
 
+### core atoms updated
```
Comment (real34) [inline source/docs/appendices/migration-guides.md:90]:
```suggestion
### Core Form atoms updated
```

**Review by real34 — APPROVED** [review]
This looks good, thanks!

Please note that we also need to document the Smart Forms feature (available inputs and how to use them) in a new documentation page.

**Comment by mo-fc** [comment]
Added docs for the smart forms module. @dpobel you can add the documentation for your email validation hook in `source/docs/advanced/features/smart-forms.md` and then it should be good

**Review by dpobel — APPROVED** [review]
@mo-fc I'm about to start documenting the email validation but no need to block this PR, I'll create a new one dedicated to email validation

**Comment by dpobel** [comment]
> @mo-fc I'm about to start documenting the email validation but no need to block this PR, I'll create a new one dedicated to email validation

that said, after checking out your branch, I've realized we already have https://developers.front-commerce.com/docs/advanced/theme/form.html#Custom-validations and the new page has no entry in the menu (`_config.yml`). I think the smart form page should go under Feature and the existing form page should be updated to have one or several to the smart form one.

**Comment by mo-fc** [comment]
Ok Ill merge this one then, then I'll work at @dpobel suggestions

**Pull request merged**